flocker/node/agents/ebs.py

```diff
@@ -276,6 +278,28 @@ class AttachFailed(Exception):
     """
 
 
+class VolumeBusy(Exception):
+    """
+    A volume could not be detached, due to being in busy state.
+    """
+    def __init__(self, volume):
+        Exception.__init__(self, volume.id, volume.attachments)
+        logged_attachments = []
+        for attachment in volume.attachments:
+            logged_attachments.append(
+                dict(
+                    instance_id=attachment['InstanceId'],
+                    volume_id=attachment['VolumeId'],
+                    device=attachment['Device'],
+                    state=attachment['State'],
+                )
+            )
+        VOLUME_BUSY_MESSAGE(
+            volume_id=volume.id,
+            attachments=logged_attachments
+        ).write()
```

Comment:
Please split the logging part of this out of the exception, and instead explicitly log before raising it. 

It's generally better to avoid doing IO in constructors, as it prevents the object in question from being used as a value. (Granted, this is a relatively theoretical consideration with this exception).